### ISSUE TITLE
Plugin messes up the results of running suite in Eclipse tab for parallel run

### ISSUE THREAD

**Issue by vkhromenkov** [cbeust/testng-eclipse#304]
### Plugin Version
6.10.0.201612030230

I have xml test suite such this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Acceptance Tests" parallel="tests" thread-count="2">
	<test name="Test Name 1">
		<classes>
			<class name="com.some.SomeClass1" />
		</classes>
	</test>
	<test name="Test Name 2">
		<classes>
			<class name="com.some.SomeClass2" />
		</classes>
	</test>
	<test name="Test Name 3">
		<classes>
			<class name="com.some.SomeClass3" />
		</classes>
	</test>
</suite>
```
### Expected behaviour
Correct test result view as with one thread

### Actual behaviour
Everytime I run this suite using Run with TestNG - results are messed up - methods from one class are in another, duplicate methods and such (for example, it shows Tests 17/6 Methods 51, but I have 6 tests with 12 methods in them)
If I run suite outside of eclipse - TestNG generates correct report and all looks fine

### Any relate message in "Error Log" view
Nothing

### The Dependency Management tool for your project
Gradle

### Operating System
Windows

**Comment by missedone** [comment]
@vkhromenkov, I can't reproduce the issue, could you share a sample project for it?

before that, do you use any ITestNGListener?

also, could you re-run the test with `verbose` option enabled: click "Run" on top level menu -> "Run Configurations..." -> locate to your launch configuration -> on the "Test" tab, scroll down to "Runtime" group, enable "Verbose" -> "Apply" -> "Run".
you will see some logs as below:
```
[JsonMessageSender] Sending message [TestMessage ==> suite:Acceptance Tests, testName:Test Name 2, passed:0, failed:0]
[JsonMessageSender] Sending message [TestMessage ==> suite:Acceptance Tests, testName:Test Name 1, passed:0, failed:0]
[JsonMessageSender] Sending message [TestResultMessage ==> suite:Acceptance Tests, test:Test Name 1, class:com.some.SomeClass1, method:testSomeClass1,
 parameters:]
[JsonMessageSender] Sending message [TestResultMessage ==> suite:Acceptance Tests, test:Test Name 2, class:com.some.SomeClass2, method:testSomeClass2,
...
```

basically, check the logs start with `[JsonMessageSender]`, and see if the message messed up or not.
thanks

**Comment by vkhromenkov** [comment]
I have tried to find the root of problem, but found some other interesting thing - this problem persists even if i have parallel=false.
So, my suite looks like
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="Acceptance Tests" parallel="false" thread-count="1">
	<test name="Data Quality">
		<classes>
			<class name="com.test.data.DataQuality" />
		</classes>
	</test>
	<test name="Delete Record">
		<classes>
			<class name="com.test.data.DeleteRecord" />
		</classes>
	</test>
</suite>
```
I'm using only one listener class (@Listeners) extending TestListenerAdapter
it's very simple, smth like this
```java
public class ScreenshotOnFailure extends TestListenerAdapter {
    @Override
    public void onTestFailure(ITestResult result) {
        Reporter.setCurrentTestResult(result);
        Date date = new Date();
        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy HH:mm:ss");
        getScreenshot(result.getTestClass().getRealClass().getSimpleName(), result.getName(), dateFormat.format(date));
        Reporter.setCurrentTestResult(null);
    }

    @Override
    public void onConfigurationFailure(ITestResult result) {
        onTestFailure(result);
    }
}
```
And TestNG verbose mode messages (yes, they are messed up too):
http://pastebin.com/Sxestgwb
Eclipse screen:
http://imgur.com/a/53N3z

I have 2 tests with 1 test method in each.

If it's not enough for investigation I will try to prepare sample project

**Comment by missedone** [comment]
hi @vkhromenkov, I don't the see message are messed up in your logs, and the class name and methods looks reasonable...
```
[JsonMessageSender] Sending message [TestResultMessage ==> suite:Acceptance Tests, test:Data Quality, class:com.test.data.DataQuality, method:checkRules, parameters:]
...
[JsonMessageSender] Sending message [TestResultMessage ==> suite:Acceptance Tests, test:Delete Record, class:com.test.data.DeleteRecord, method:delete, parameters:]
```

could you share a sample project for this?

**Comment by vkhromenkov** [comment]
I see several instances of the same message, e.g.
[JsonMessageSender] Sending message [TestResultMessage ==> suite:Acceptance Tests, test:Delete Record, class:com.test.data.DeleteRecord, method:delete, parameters:]

And as a result I see 2 instances of method delete in visual report (as you can see on screenshot), but I expect only one

Furthermore - I don't think "Tests: 3/2" is correct text for test result (you can see it on screenshot too)

**Comment by vkhromenkov** [comment]
I have prepared sample project - you can get it here https://github.com/vkhromenkov/sel-1-repo
There is no need for any libs - just several dummy tests and acceptance.xml to run them.
Just ignore other stuff.
So, open it in eclipse and run this xml suite as TestNG suite.
My run looks really weird - http://imgur.com/a/nmr62

**Comment by missedone** [comment]
looks like issue solved if you update to TestNG 6.10 in the build.gradle: `testCompile 'org.testng:testng:6.10'`, below is the result:

<img width="1003" alt="screenshot at jan 01 04-27-32" src="https://cloud.githubusercontent.com/assets/555134/21581187/0a9846b0-cfdb-11e6-85f4-2b126fcc92e5.png">

**Comment by vkhromenkov** [comment]
Sorry for long wait - was on vacation. 
Yes, updating testNG jar to 6.10 fixed the problem.
But I don't understand how testNG jar (in project) and testNG eclipse plugin interact with each other.
I thought plugin is responsible for running tests inside ide while jar - for running tests outside of it.
Maybe the problem was that I had 6.9 jar and 6.10 plugin

**Comment by missedone** [comment]
the plugin starts a new JVM for running the tests with TestNG jar on the runtime classpath.
the runtime TestNG process will send back test result to plugin, then the plugin will display the them on the result viewer.
it's an issue came from TestNG jar.
